src/index.js

```diff
@@ -1,32 +1,27 @@
-'use strict';
-
-var xfilterArray = require('./array');
-var xfilterFilter = require('./filter');
-var crossfilter_identity = require('./identity');
-var crossfilter_null = require('./null');
-var crossfilter_zero = require('./zero');
-var xfilterHeapselect = require('./heapselect');
-var xfilterHeap = require('./heap');
-var bisect = require('./bisect');
-var insertionsort = require('./insertionsort');
-var permute = require('./permute');
-var quicksort = require('./quicksort');
-var xfilterReduce = require('./reduce');
-var packageJson = require('./../package.json'); // require own package.json for the version field
-var result = require('lodash.result');
+import xfilterArray from './array';
+import xfilterFilter from './filter';
+import cr_identity from './identity';
+import cr_null from './null';
+import cr_zero from './zero';
+import xfilterHeapselect from './heapselect';
+import xfilterHeap from './heap';
+import bisect from './bisect';
+import insertionsort from './insertionsort';
+import permute from './permute';
+import quicksort from './quicksort';
+import xfilterReduce from './reduce';
+import result from './result';
 
 // constants
 var REMOVED_INDEX = -1;
 
-// expose API exports
-exports.crossfilter = crossfilter;
-exports.crossfilter.heap = xfilterHeap;
-exports.crossfilter.heapselect = xfilterHeapselect;
-exports.crossfilter.bisect = bisect;
-exports.crossfilter.insertionsort = insertionsort;
-exports.crossfilter.permute = permute;
-exports.crossfilter.quicksort = quicksort;
-exports.crossfilter.version = packageJson.version; // please note use of "package-json-versionify" transform
```

Comment:
Is there no other way to make it possible for the users to request the crossfilter version at runtime?

It probably isn't used that much, but still nice to have IMHO.

Reply:
Tried hard on that, but nope, no simple solution. We might inject at build time if deemed necessary.

Reply:
@gordonwoodhull, @esjewett - what do you think?

Reply:
Good catch, @TijmenW!

Yes, the version number is essential for helping random people on the internet - much easier to look at this property than to try grepping the source for things which have changed, e.g. when trying to determine if someone might be affected by a fixed bug. (It helps as a sanity check for my own build processes too.)

A quick search reveals this technique:

```js
import { version } from '../package.json';
crossfilter.version = version;
```

Pretty simple but I wouldn't have thought of it myself!

Reply:
Tried that - won't work in browsers, you get 
```
Failed to load module script: The server responded with a non-JavaScript MIME type of "application/json". Strict MIME type checking is enforced for module scripts per HTML spec.
``` 

https://stackoverflow.com/questions/34944099/how-to-import-a-json-file-in-ecmascript-6

Reply:
Aha, I only tested it in the rollup/UMD case. I'm not sure what we need to do to make it work in the ES6 module case. 

Personally I would be okay with only importing/exporting the version in the UMD case, since anyone using the ES6 modules should have the entire npm'd source tree with package.json.

(For that matter, the function-as-module trick isn't needed in ES6; it would be more idiomatic, and probably better for tree shaking, not to add any extra fields to the crossfilter function. But we can think about that later.)

Reply:
6198988 now exports version to UMD build